spec/controllers/bills_controller_spec.rb

```diff
@@ -0,0 +1,72 @@
+require 'rails_helper'
+
+describe BillsController do
+
+  before(:each) { @bill = create(:bill) }
+
+  describe 'GET #index' do
+    before { get :index }
+
+    it { is_expected.to respond_with :ok }
+    it { is_expected.to render_template :index }
+    it { expect(@bill).to_not be nil }
+  end
+
+  describe "GET #new" do
+    before { get :new }
+
+    it { is_expected.to respond_with :ok }
+    it { is_expected.to render_template :new }
+  end
+
+  describe "POST #create" do
+
+    it "should create bill" do
+      user = create(:user)
+      sign_in user
+      bill = FactoryGirl.attributes_for(:bill).merge(user_id: user.id, units: {number_occupants: 4})
+
+      expect{ post(:create, bill: bill) }.to change{ Bill.count }.by(1)
+    end
+  end
+
+  describe "GET #edit" do
+    before { get :edit, id: @bill }
+
+    it { is_expected.to respond_with :ok }
+    it { is_expected.to render_template :edit }
+  end
+
+
+  describe "GET #comarison" do
+    before do
+      user = create(:user)
+      sign_in user
+      get :comparison
+    end
+
+    it { is_expected.to render_template :comparison }
+  end
+
+  describe "POST #update" do
+    before do
```

Comment:
Since we only have 1 assertion, we don't need to have a `before` block. Let's do something like the following instead:

```ruby
describe "PUT #update" do
  it "redirects to the bill show page" do
    patch :update, id: @bill.id, bill: { ... }

    expect(response).to redirect_to bill_path(@bill)
  end
end
```